compiler-core/src/format.rs

```diff
@@ -1529,13 +1531,24 @@ impl<'comments> Formatter<'comments> {
         }
     }
 
-    fn negate<'a>(&mut self, expr: &'a UntypedExpr) -> Document<'a> {
+    fn negate_bool<'a>(&mut self, expr: &'a UntypedExpr) -> Document<'a> {
         match expr {
             UntypedExpr::BinOp { .. } => docvec!["!{ ", self.expr(expr), " }"],
             _ => docvec!["!", self.wrap_expr(expr)],
         }
     }
 
+    fn negate_int<'a>(&mut self, expr: &'a UntypedExpr) -> Document<'a> {
+        match expr {
+            // Always nest repeated negation in a block to avoid confusion with
+            // the pre-decrement operator (which does not exist)
+            UntypedExpr::BinOp { .. } | UntypedExpr::NegateInt { .. } => {
+                docvec!["- { ", self.expr(expr), " }"]
```

Comment:
This unfortunately won't wrap and indent correctly when the line is too long. Let's extract the block wrapping code from `wrap_expr` into a helper function and use that here.

Reply:
I ended up cribbing from `operator_side`, `wrap_expr` includes some newlines that weren't always necessary.